### PR TITLE
PTX-13087 Run pxctl command with Security token if PX Security is enabled

### PR DESCRIPTION
**Pull request by nikolaypopov**
Run pxctl command with Security token if PX Security is enabled

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -3056,22 +3056,30 @@ func ValidateTelemetryV2Enabled(pxImageList map[string]string, cluster *corev1.S
...
 	logrus.Infof("All Telemetry components were successfully enabled/installed")
 	return nil
 }
 
-func validateTelemetryStatusInPxctl(telemetryShouldBeEnabled bool, namespace string) error {
+func validateTelemetryStatusInPxctl(telemetryShouldBeEnabled bool, cluster *corev1.StorageCluster) error {
 	listOptions := map[string]string{"name": "portworx"}
 	cmd := "pxctl status | grep Telemetry:"
 
+	if cluster.Spec.Security != nil {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:3071]: && cluster.Spec.Security.Enabled
and move this inside of the retry block.

File: pkg/util/test/util.go [hunk 2]
```diff
@@ -3107,6 +3115,18 @@ func validateTelemetryStatusInPxctl(telemetryShouldBeEnabled bool, namespace str
 	return nil
 }
 
+func getSecurityAdminToken(namespace string) (string, error) {
+	logrus.Debug("PX Security is enabled, getting token from px-admin-token secret")
+
+	secret, err := coreops.Instance().GetSecret("px-admin-token", namespace)
```
Comment (pure-jliao) [inline pkg/util/test/util.go:3121]: does this retry if network issue?

File: test/integration_test/basic_test.go [hunk 3]
```diff
@@ -492,7 +492,7 @@ func BasicTelemetryRegression(tc *types.TestCase) func(*testing.T) {
 		if cluster.Spec.Monitoring != nil {
 			if cluster.Spec.Monitoring.Telemetry != nil {
 				if cluster.Spec.Monitoring.Telemetry.Enabled {
```
Comment (pure-jliao) [inline test/integration_test/basic_test.go:494]: nit: we can remove this condition check

File: pkg/util/test/util.go (hunk 1 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go]: Done

File: pkg/util/test/util.go (hunk 2 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:3121]: I think its enough for it to be inside one retry loop

File: test/integration_test/basic_test.go (hunk 3 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go:494]: I will leave it for now as we are validating that its not enabled by default

Comment (pure-jliao) [reply test/integration_test/basic_test.go:494]: yes, but if it's enabled, it still return error. it's a duplicated condition check. but I'm ok with that.

**Review by pure-jliao — APPROVED** [review]

File: test/integration_test/basic_test.go (hunk 3 above)
Comment (nikolaypopov) [reply test/integration_test/basic_test.go]: I misunderstood what you said, done ^^

**Pull request merged**